### DISCUSSION TITLE
Consider binary UUID storage beyond Postgres

### DISCUSSION THREAD

**Discussion by mostlyobvious** [RailsEventStore/rails_event_store#1695] [Ideas]
If it helps in any way for performance let's brainstorm how we could introduce it.

**Comment by mostlyobvious** [comment]
Related: https://github.com/the-cave/event-id-mapper

**Comment by mostlyobvious** [comment]
Related: https://github.com/nedap/mysql-binuuid-rails

**Comment by AlexeyMatskevich** [comment]
Hi, is it actual for rails 6? [Rails 6 added uuid](https://guides.rubyonrails.org/active_record_postgresql.html#uuid-primary-keys).

**Comment by mostlyobvious** [comment]
It is true that ActiveRecord schema DSL  supports uuids, and it was true even before Rails 6. Does Rails 6 extend it beyond Postgres adapter?

In scope of storage RailsEventStore has to consider database support as well. This is not only Postgres but also MySQL and Sqlite.

There has been addition of native uuids in Mysql 8.0, which would be a nice improvement if that worked out of the box in ActiveRecord. Then we'd have to only adjust schema generation for this particular database version in RES.

**Comment by mostlyobvious** [comment]
(renamed title to be more precise)
